crates/pallets/starknet/src/utils.rs

```diff
@@ -48,3 +48,50 @@ pub fn execute_txs_and_rollback<T: pallet::Config>(
 
     Ok(execution_results)
 }
+
+pub fn execute_txs_until_first_failed_and_rollback<T: pallet::Config>(
+    txs: &Vec<UserTransaction>,
+    block_context: &BlockContext,
+    chain_id: Felt252Wrapper,
+    execution_config: &mut ExecutionConfig,
+) -> Result<Result<Vec<TransactionExecutionInfo>, PlaceHolderErrorTypeForFailedStarknetExecution>, Error<T>> {
+    let mut execution_infos = vec![];
+
+    storage::transactional::with_transaction(|| {
+        for tx in txs {
+            execution_config.set_offset_version(tx.offset_version());
+            let result = match tx {
+                UserTransaction::Declare(tx, contract_class) => tx
+                    .try_into_executable::<T::SystemHash>(chain_id, contract_class.clone(), tx.offset_version())
+                    .and_then(|exec| {
+                        exec.execute(&mut BlockifierStateAdapter::<T>::default(), block_context, execution_config)
+                    }),
+                UserTransaction::DeployAccount(tx) => {
+                    let executable = tx.into_executable::<T::SystemHash>(chain_id, tx.offset_version());
+                    executable.execute(&mut BlockifierStateAdapter::<T>::default(), block_context, execution_config)
+                }
+                UserTransaction::Invoke(tx) => {
+                    let executable = tx.into_executable::<T::SystemHash>(chain_id, tx.offset_version());
+                    executable.execute(&mut BlockifierStateAdapter::<T>::default(), block_context, execution_config)
+                }
+            };
+
+            // In case of revert (cairo1) or failed assertion (cairo0) we stop iterating
+            // For now we don't know how to move the revert_reason out of the runtime, so we can keep things
+            // simple with the error handling
+            match result {
+                Ok(tx_info) if !tx_info.is_reverted() => execution_infos.push(tx_info),
+                _ => break,
```

Comment:
I don't think we should simply break but rather return early with an error in the rollback no?

Reply:
Also we might want to simply use the transactional [macro](https://paritytech.github.io/polkadot-sdk/master/frame_support/attr.transactional.html) wdyt ?

Reply:
> I don't think we should simply break but rather return early with an error in the rollback no?

We need to execute this line of code `storage::TransactionOutcome::Rollback(Result::<_, DispatchError>::Ok(()))`. Which says "I successfully executed what I needed to execut

Reply:
> Also we might want to simply use the transactional [macro](https://paritytech.github.io/polkadot-sdk/master/frame_support/attr.transactional.html) wdyt ?

For transactional function we need to return an error in order to revert storage changes. But we also need to get a vec execution infos out of this. Both can't be done

Reply:
> For transactional function we need to return an error in order to revert storage changes. But we also need to get a vec execution infos out of this. Both can't be done


I was wrong! It is doable. And it's in the new commits